File: src/leetcode/python/WordBreak_139.py
class WordBreak_139(object):
    def wordBreak(self, s, wordDict):
        """
        :type s: str
        :type wordDict: List[str]
        :rtype: bool
        """
        l = len(s)+1
        dp = [False] * l
        dp[0] = True
        for i in range(1, l):
            for j in range(0, i):
                dp[i] |= dp[j] and s[j:i] in wordDict
        return dp[len(s)]

File: src/leetcode/java/WordBreak_139.java
package leetcode.java;

import java.util.List;

public class WordBreak_139 {
  public boolean wordBreak(String s, List<String> wordDict) {
    boolean[] dp = new boolean[s.length() + 1];
    dp[0] = true;
    for (int i = 1; i <= s.length(); i++) {
      for (int j = 0; j < i; j++) {
        dp[i] |= dp[j] && wordDict.contains(s.substring(j, i));
      }
    }
    return dp[s.length()];
  }
}
